crates/pkvdb/src/lib.rs

```diff
@@ -0,0 +1,451 @@
+use hash_db::{AsHashDB, AsPlainDB, HashDB, HashDBRef, Hasher, PlainDB, PlainDBRef, Prefix};
+use rusty_leveldb::gramine_env::GramineEnv;
+use rusty_leveldb::{LdbIterator, Options as LevelDBOptions, WriteBatch, DB};
+use sp_state_machine::backend::Consolidate;
+pub use sp_trie::MemoryDB as Transaction;
+use std::borrow::BorrowMut;
+use std::ops::DerefMut;
+use std::path::Path;
+use std::rc::Rc;
+use std::sync::Arc;
+use std::sync::Mutex;
+
+pub type LevelDB<H> = Kvdb<H, sp_trie::DBValue>;
+
+pub struct Kvdb<H, T>
+where
+    H: Hasher,
+{
+    leveldb: Arc<Mutex<DB>>,
```

Comment:
Would we have a seperate type like `KvdbSnapshot`?

Reply:
underlying KVDB had supported the snapshot, everytime R/W will take a snapshot firsly, We do not care about any logic with snapshot